Test setup failures with Bash 5.1: AssertionError: '\x1b[?2004l\r\x1b[?2004h' != ''

Hello. In Fedora Rawhide (development branch), we have updated to Bash 5.1. The tests here fail to setup with:

```
======================================================================
FAIL: test_comp_point (test.test.TestBash)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-1.12.0/test/test.py", line 1206, in setUp
    self.assertEqual(output, '')
AssertionError: '\x1b[?2004l\r\x1b[?2004h' != ''
- 

======================================================================
FAIL: test_completion_environment (test.test.TestBash)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-1.12.0/test/test.py", line 1206, in setUp
    self.assertEqual(output, '')
AssertionError: '\x1b[?2004l\r\x1b[?2004h' != ''
- 

======================================================================
FAIL: test_continuation (test.test.TestBash)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-1.12.0/test/test.py", line 1206, in setUp
    self.assertEqual(output, '')
AssertionError: '\x1b[?2004l\r\x1b[?2004h' != ''
- 

======================================================================
FAIL: test_debug_output (test.test.TestBash)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-1.12.0/test/test.py", line 1206, in setUp
    self.assertEqual(output, '')
AssertionError: '\x1b[?2004l\r\x1b[?2004h' != ''
- 

======================================================================
FAIL: test_double_quoted_completion (test.test.TestBash)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-1.12.0/test/test.py", line 1206, in setUp
    self.assertEqual(output, '')
AssertionError: '\x1b[?2004l\r\x1b[?2004h' != ''
- 

======================================================================
FAIL: test_exclamation_in_double_quotes (test.test.TestBash)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-1.12.0/test/test.py", line 1206, in setUp
    self.assertEqual(output, '')
AssertionError: '\x1b[?2004l\r\x1b[?2004h' != ''
- 

======================================================================
FAIL: test_one_space_after_exact (test.test.TestBash)
Test exactly one space is appended after an exact match.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-1.12.0/test/test.py", line 1206, in setUp
    self.assertEqual(output, '')
AssertionError: '\x1b[?2004l\r\x1b[?2004h' != ''
- 

======================================================================
FAIL: test_parse_special_characters (test.test.TestBash)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-1.12.0/test/test.py", line 1206, in setUp
    self.assertEqual(output, '')
AssertionError: '\x1b[?2004l\r\x1b[?2004h' != ''
- 

======================================================================
FAIL: test_parse_special_characters_dollar (test.test.TestBash)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-1.12.0/test/test.py", line 1206, in setUp
    self.assertEqual(output, '')
AssertionError: '\x1b[?2004l\r\x1b[?2004h' != ''
- 

======================================================================
FAIL: test_partial_completion (test.test.TestBash)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-1.12.0/test/test.py", line 1206, in setUp
    self.assertEqual(output, '')
AssertionError: '\x1b[?2004l\r\x1b[?2004h' != ''
- 

======================================================================
FAIL: test_quoted_exact (test.test.TestBash)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-1.12.0/test/test.py", line 1206, in setUp
    self.assertEqual(output, '')
AssertionError: '\x1b[?2004l\r\x1b[?2004h' != ''
- 

======================================================================
FAIL: test_quoted_space (test.test.TestBash)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-1.12.0/test/test.py", line 1206, in setUp
    self.assertEqual(output, '')
AssertionError: '\x1b[?2004l\r\x1b[?2004h' != ''
- 

======================================================================
FAIL: test_quotes (test.test.TestBash)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-1.12.0/test/test.py", line 1206, in setUp
    self.assertEqual(output, '')
AssertionError: '\x1b[?2004l\r\x1b[?2004h' != ''
- 

======================================================================
FAIL: test_simple_completion (test.test.TestBash)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-1.12.0/test/test.py", line 1206, in setUp
    self.assertEqual(output, '')
AssertionError: '\x1b[?2004l\r\x1b[?2004h' != ''
- 

======================================================================
FAIL: test_single_quoted_completion (test.test.TestBash)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-1.12.0/test/test.py", line 1206, in setUp
    self.assertEqual(output, '')
AssertionError: '\x1b[?2004l\r\x1b[?2004h' != ''
- 

======================================================================
FAIL: test_single_quotes_in_double_quotes (test.test.TestBash)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-1.12.0/test/test.py", line 1206, in setUp
    self.assertEqual(output, '')
AssertionError: '\x1b[?2004l\r\x1b[?2004h' != ''
- 

======================================================================
FAIL: test_single_quotes_in_single_quotes (test.test.TestBash)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-1.12.0/test/test.py", line 1206, in setUp
    self.assertEqual(output, '')
AssertionError: '\x1b[?2004l\r\x1b[?2004h' != ''
- 

======================================================================
FAIL: test_special_characters (test.test.TestBash)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-1.12.0/test/test.py", line 1206, in setUp
    self.assertEqual(output, '')
AssertionError: '\x1b[?2004l\r\x1b[?2004h' != ''
- 

======================================================================
FAIL: test_special_characters_double_quoted (test.test.TestBash)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-1.12.0/test/test.py", line 1206, in setUp
    self.assertEqual(output, '')
AssertionError: '\x1b[?2004l\r\x1b[?2004h' != ''
- 

======================================================================
FAIL: test_special_characters_single_quoted (test.test.TestBash)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-1.12.0/test/test.py", line 1206, in setUp
    self.assertEqual(output, '')
AssertionError: '\x1b[?2004l\r\x1b[?2004h' != ''
- 

======================================================================
FAIL: test_temp_file (test.test.TestBash)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-1.12.0/test/test.py", line 1206, in setUp
    self.assertEqual(output, '')
AssertionError: '\x1b[?2004l\r\x1b[?2004h' != ''
- 

======================================================================
FAIL: test_unquoted_space (test.test.TestBash)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-1.12.0/test/test.py", line 1206, in setUp
    self.assertEqual(output, '')
AssertionError: '\x1b[?2004l\r\x1b[?2004h' != ''
- 

======================================================================
FAIL: test_wordbreak_chars (test.test.TestBash)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-1.12.0/test/test.py", line 1206, in setUp
    self.assertEqual(output, '')
AssertionError: '\x1b[?2004l\r\x1b[?2004h' != ''
- 

======================================================================
FAIL: test_comp_point (test.test.TestBashGlobal)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-1.12.0/test/test.py", line 1206, in setUp
    self.assertEqual(output, '')
AssertionError: '\x1b[?2004l\r\x1b[?2004h' != ''
- 

======================================================================
FAIL: test_completion_environment (test.test.TestBashGlobal)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-1.12.0/test/test.py", line 1206, in setUp
    self.assertEqual(output, '')
AssertionError: '\x1b[?2004l\r\x1b[?2004h' != ''
- 

======================================================================
FAIL: test_console_script_module (test.test.TestBashGlobal)
Test completing a console_script for a module.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-1.12.0/test/test.py", line 1206, in setUp
    self.assertEqual(output, '')
AssertionError: '\x1b[?2004l\r\x1b[?2004h' != ''
- 

======================================================================
FAIL: test_console_script_module_wheel (test.test.TestBashGlobal)
Test completing a console_script for a module from a wheel.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-1.12.0/test/test.py", line 1206, in setUp
    self.assertEqual(output, '')
AssertionError: '\x1b[?2004l\r\x1b[?2004h' != ''
- 

======================================================================
FAIL: test_console_script_package (test.test.TestBashGlobal)
Test completing a console_script for a package.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-1.12.0/test/test.py", line 1206, in setUp
    self.assertEqual(output, '')
AssertionError: '\x1b[?2004l\r\x1b[?2004h' != ''
- 

======================================================================
FAIL: test_console_script_package_wheel (test.test.TestBashGlobal)
Test completing a console_script for a package from a wheel.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-1.12.0/test/test.py", line 1206, in setUp
    self.assertEqual(output, '')
AssertionError: '\x1b[?2004l\r\x1b[?2004h' != ''
- 

======================================================================
FAIL: test_continuation (test.test.TestBashGlobal)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-1.12.0/test/test.py", line 1206, in setUp
    self.assertEqual(output, '')
AssertionError: '\x1b[?2004l\r\x1b[?2004h' != ''
- 

======================================================================
FAIL: test_debug_output (test.test.TestBashGlobal)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-1.12.0/test/test.py", line 1206, in setUp
    self.assertEqual(output, '')
AssertionError: '\x1b[?2004l\r\x1b[?2004h' != ''
- 

======================================================================
FAIL: test_double_quoted_completion (test.test.TestBashGlobal)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-1.12.0/test/test.py", line 1206, in setUp
    self.assertEqual(output, '')
AssertionError: '\x1b[?2004l\r\x1b[?2004h' != ''
- 

======================================================================
FAIL: test_exclamation_in_double_quotes (test.test.TestBashGlobal)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-1.12.0/test/test.py", line 1206, in setUp
    self.assertEqual(output, '')
AssertionError: '\x1b[?2004l\r\x1b[?2004h' != ''
- 

======================================================================
FAIL: test_one_space_after_exact (test.test.TestBashGlobal)
Test exactly one space is appended after an exact match.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-1.12.0/test/test.py", line 1206, in setUp
    self.assertEqual(output, '')
AssertionError: '\x1b[?2004l\r\x1b[?2004h' != ''
- 

======================================================================
FAIL: test_parse_special_characters (test.test.TestBashGlobal)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-1.12.0/test/test.py", line 1206, in setUp
    self.assertEqual(output, '')
AssertionError: '\x1b[?2004l\r\x1b[?2004h' != ''
- 

======================================================================
FAIL: test_parse_special_characters_dollar (test.test.TestBashGlobal)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-1.12.0/test/test.py", line 1206, in setUp
    self.assertEqual(output, '')
AssertionError: '\x1b[?2004l\r\x1b[?2004h' != ''
- 

======================================================================
FAIL: test_partial_completion (test.test.TestBashGlobal)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-1.12.0/test/test.py", line 1206, in setUp
    self.assertEqual(output, '')
AssertionError: '\x1b[?2004l\r\x1b[?2004h' != ''
- 

======================================================================
FAIL: test_python_completion (test.test.TestBashGlobal)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-1.12.0/test/test.py", line 1206, in setUp
    self.assertEqual(output, '')
AssertionError: '\x1b[?2004l\r\x1b[?2004h' != ''
- 

======================================================================
FAIL: test_python_filename_completion (test.test.TestBashGlobal)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-1.12.0/test/test.py", line 1206, in setUp
    self.assertEqual(output, '')
AssertionError: '\x1b[?2004l\r\x1b[?2004h' != ''
- 

======================================================================
FAIL: test_python_module (test.test.TestBashGlobal)
Test completing a module run with python3 -m.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-1.12.0/test/test.py", line 1206, in setUp
    self.assertEqual(output, '')
AssertionError: '\x1b[?2004l\r\x1b[?2004h' != ''
- 

======================================================================
FAIL: test_python_not_executable (test.test.TestBashGlobal)
Test completing a script that cannot be run directly.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-1.12.0/test/test.py", line 1206, in setUp
    self.assertEqual(output, '')
AssertionError: '\x1b[?2004l\r\x1b[?2004h' != ''
- 

======================================================================
FAIL: test_quoted_exact (test.test.TestBashGlobal)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-1.12.0/test/test.py", line 1206, in setUp
    self.assertEqual(output, '')
AssertionError: '\x1b[?2004l\r\x1b[?2004h' != ''
- 

======================================================================
FAIL: test_quoted_space (test.test.TestBashGlobal)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-1.12.0/test/test.py", line 1206, in setUp
    self.assertEqual(output, '')
AssertionError: '\x1b[?2004l\r\x1b[?2004h' != ''
- 

======================================================================
FAIL: test_quotes (test.test.TestBashGlobal)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-1.12.0/test/test.py", line 1206, in setUp
    self.assertEqual(output, '')
AssertionError: '\x1b[?2004l\r\x1b[?2004h' != ''
- 

======================================================================
FAIL: test_simple_completion (test.test.TestBashGlobal)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-1.12.0/test/test.py", line 1206, in setUp
    self.assertEqual(output, '')
AssertionError: '\x1b[?2004l\r\x1b[?2004h' != ''
- 

======================================================================
FAIL: test_single_quoted_completion (test.test.TestBashGlobal)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-1.12.0/test/test.py", line 1206, in setUp
    self.assertEqual(output, '')
AssertionError: '\x1b[?2004l\r\x1b[?2004h' != ''
- 

======================================================================
FAIL: test_single_quotes_in_double_quotes (test.test.TestBashGlobal)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-1.12.0/test/test.py", line 1206, in setUp
    self.assertEqual(output, '')
AssertionError: '\x1b[?2004l\r\x1b[?2004h' != ''
- 

======================================================================
FAIL: test_single_quotes_in_single_quotes (test.test.TestBashGlobal)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-1.12.0/test/test.py", line 1206, in setUp
    self.assertEqual(output, '')
AssertionError: '\x1b[?2004l\r\x1b[?2004h' != ''
- 

======================================================================
FAIL: test_special_characters (test.test.TestBashGlobal)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-1.12.0/test/test.py", line 1206, in setUp
    self.assertEqual(output, '')
AssertionError: '\x1b[?2004l\r\x1b[?2004h' != ''
- 

======================================================================
FAIL: test_special_characters_double_quoted (test.test.TestBashGlobal)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-1.12.0/test/test.py", line 1206, in setUp
    self.assertEqual(output, '')
AssertionError: '\x1b[?2004l\r\x1b[?2004h' != ''
- 

======================================================================
FAIL: test_special_characters_single_quoted (test.test.TestBashGlobal)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-1.12.0/test/test.py", line 1206, in setUp
    self.assertEqual(output, '')
AssertionError: '\x1b[?2004l\r\x1b[?2004h' != ''
- 

======================================================================
FAIL: test_temp_file (test.test.TestBashGlobal)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-1.12.0/test/test.py", line 1206, in setUp
    self.assertEqual(output, '')
AssertionError: '\x1b[?2004l\r\x1b[?2004h' != ''
- 

======================================================================
FAIL: test_unquoted_space (test.test.TestBashGlobal)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-1.12.0/test/test.py", line 1206, in setUp
    self.assertEqual(output, '')
AssertionError: '\x1b[?2004l\r\x1b[?2004h' != ''
- 

======================================================================
FAIL: test_wordbreak_chars (test.test.TestBashGlobal)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/argcomplete-1.12.0/test/test.py", line 1206, in setUp
    self.assertEqual(output, '')
AssertionError: '\x1b[?2004l\r\x1b[?2004h' != ''
- 

----------------------------------------------------------------------
Ran 144 tests in 84.144s

FAILED (failures=54, skipped=2, expected failures=9)

```

I unfortunately fail to reproduce this outside of our build environment :(

We have 1.12.0.

--- a/test/inputrc
+++ b/test/inputrc
@@ -1,0 +1,1 @@
+set enable-bracketed-paste off

--- a/test/test.py
+++ b/test/test.py
@@ -3,6 +3,7 @@
 import argparse
 import contextlib
 import os
+import os.path
 import shutil
 import subprocess
 import sys
@@ -38,6 +39,10 @@ BASH_VERSION = subprocess.check_output(["bash", "-c", "echo $BASH_VERSION"]).dec
 BASH_MAJOR_VERSION = int(BASH_VERSION.split(".")[0])
 FISH_VERSION_STR = subprocess.check_output(["fish", "-c", "echo -n $version"]).decode()
 FISH_VERSION_TUPLE = tuple(int(x) for x in FISH_VERSION_STR.split("."))
+
+
+def setUpModule():
+    os.environ['INPUTRC'] = os.path.join(os.path.dirname(__file__), 'inputrc')
 
 
 class TempDir(object):

